src/main/java/org/opensearch/geospatial/ip2geo/dao/Ip2GeoCachedDao.java

```diff
@@ -112,9 +118,26 @@ private void remove(final String datasourceName) {
         getMetadata().remove(datasourceName);
     }
 
+    private void clearMetadata() {
+        log.info("Resetting all datasource metadata to force a refresh from the primary index shard.");
+        metadata = null;
```

Comment:
We might need to handle it more gracefully. Current metadata reading shouldn't be breaking due to this code.

Reply:
Consider initializing an empty Map instead？ Will it cause NPE when getMetadata() is called elsewhere?